src/net/sourceforge/kolmafia/swingui/widget/ShowDescriptionList.java

```diff
@@ -320,6 +323,28 @@ protected void executeAction() {
     }
   }
 
+  public class ForbidStoreRunnable extends ContextMenuListener {
+    @Override
+    protected void executeAction() {
+      if (!(this.item instanceof PurchaseRequest)) {
+        return;
+      }
+
+      List<String> forbidden = MallPurchaseRequest.getForbiddenStores();
+      String storeId = ((PurchaseRequest) this.item).getFormField("whichstore");
+
+      if (storeId == null || !storeId.matches("[0-9]+")) {
```

Comment:
I think the code here could be simplified a bit by converting to an int here and catching numberformatexception